### PR TITLE
Add Comments tab to submission

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-12109
affects: none
release note: none

### CODE REVIEW

**Comment by vanwagonet** [comment]
Notes:
- User avatars are just a simple `UIImageView` right now, so this doesn't handle showing initials when no avatar url is present.
- Non-text comments are rendered as blank cells, so there is going to be some extra white space until those are handled.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  One or more files in Core are below the minimum test coverage 50%
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-12109](https://instructure.atlassian.net/browse/MBL-12109)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.45% | 93.36% | 0.09%
Core/Core/Extensions/OperationQueueExtensions.swift | 0% | 0% | 0%
**Student** | 94.74% | 94.68% | 0.06%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.54% | 92.54% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Database/CoreData/Models/SubmissionComment.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        model.mediaID = item.media_comment?.media_id
+        model.mediaName = item.media_comment?.display_name
+        model.mediaType = item.media_comment?.media_type
+        model.mediaURL = item.media_comment?.url
+        model.submissionID = submissionID
+        return model
+    }
+
+    @discardableResult
+    static public func save(_ item: APISubmission, in client: PersistenceClient) -> SubmissionComment? {
```
Comment (shichongrui) [inline Core/Core/Database/CoreData/Models/SubmissionComment.swift]: What is the purpose of having a comment for the attempt that is empty?

File: Student/Student/Submissions/SubmissionComments/SubmissionCommentsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        if let mediaID = comment.mediaID, let mediaURL = comment.mediaURL, let mediaType = comment.mediaType {
+            // TODO: display media comments
+            let cell = UITableViewCell()
+            cell.accessibilityLabel = "\(mediaType) \(mediaID) \(mediaURL)" // get rid of unused variable errors
+            return cell
+        }
+        if let attempt = comment.attempt {
+            // TODO: display submission attachments
+            let cell = UITableViewCell()
+            cell.accessibilityLabel = "\(attempt)" // get rid of unused variable errors
```
Comment (shichongrui) [inline Student/Student/Submissions/SubmissionComments/SubmissionCommentsViewController.swift]: Could you just change it to `if let _ = comment.attempt`? Or do you anticipate needing the attempt later? Same for above.

File: scripts/coverage/config.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
     "\\/Views\\/",
     "Core\/ShareExtension\/",
     "Core\/Use Cases\/SubmitFileSubmission",
+    "OperationQueueExtensions.swift",
```
Comment (shichongrui) [inline scripts/coverage/config.json]: haha thanks!

File: Core/Core/Database/CoreData/Models/SubmissionComment.swift (hunk 1 above)
Comment (vanwagonet) [reply Core/Core/Database/CoreData/Models/SubmissionComment.swift]: We will show a comment for each submission attempt, including a list to all the submitted files. The comment display is in another ticket, but I put the SubmissionComment entry in already. Having it as an entry with the rest of the comments makes it easy to be sorted and displayed with the rest.

Comment (shichongrui) [reply Core/Core/Database/CoreData/Models/SubmissionComment.swift]: 👍

File: Student/Student/Submissions/SubmissionComments/SubmissionCommentsViewController.swift (hunk 2 above)
Comment (vanwagonet) [reply Student/Student/Submissions/SubmissionComments/SubmissionCommentsViewController.swift]: We'll need the attempt number later to find the submission object and build a cell for it.

Comment (shichongrui) [reply Student/Student/Submissions/SubmissionComments/SubmissionCommentsViewController.swift]: 👍

**Pull request merged**